src/main/js/controls/Bubble/BubbleSingleDataset.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+
+import constants, { BUBBLE } from "../../helpers/constants";
+import { prepareLabelShapeItem } from "../../helpers/label";
+import { reflowLegend } from "../../helpers/legend";
+import {
+    createRegion,
+    hideAllRegions,
+    translateRegion,
+    areRegionsIdentical
+} from "../../helpers/region";
+import styles from "../../helpers/styles";
+import utils from "../../helpers/utils";
+import {
+    clear,
+    clickHandler,
+    hoverHandler,
+    prepareLegendItems,
+    processDataPoints,
+    translateBubbleGraph,
+    getDataPointValues,
+    calculateValuesRange
+} from "./helpers/helpers";
+import { draw, drawBubbles } from "./helpers/helpersSingleDataset";
+import Bubble from "./Bubble";
+
+/**
+ * A Bubble graph is a graph used to represent a collection of data
+ * points connected by a Bubble along the X and Y Axis.
+ *
+ * Lifecycle functions include:
+ *  * Load
+ *  * Generate
+ *  * Unload
+ *  * Destroy
+ *
+ * @module BubbleSingleDataset
+ * @class BubbleSingleDataset
+ */
+class BubbleSingleDataset extends Bubble {
+    /**
+     * @inheritdoc
+     * @public
+     */
+    load(graph) {
+        if (graph.content.length > 1) {
+            console.warn("BubbleSingleDataset can only use one dataset.");
```

Comment:
Is it worth having a warning? or Should we log an error if the consumer provides more than one dataset?

Reply:
Yes, as any subsequent datasets will be ignored and the consumer needs to know why. I can update this to a warning if necessary.